rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java

```diff
@@ -442,12 +429,12 @@ private Block createFakePendingBlock(Block best) {
                             ByteUtil.bigIntegerToBytes(BigInteger.ZERO)); //minimum gas price
     }
 
-    private boolean shouldAcceptTx(Transaction tx) {
-        if (bestBlock == null) {
+    private boolean shouldAcceptTx(Transaction tx, Block currentBestBlock) {
```

Comment:
After construction, the object variable bestBlock is null.

This leaves a window to send a pending local transaction, BEFORE this.bestBlock is filled. 

We could provide a best block known so far at constructor (at least, the genesis block)

Then, the validation of transaction could evolve and we still will have the right incoming transaction validation applied

IE, now, we could provide a transaction with gas limit out of range.

In the future, we should remove the validation of transaction at this point: it could be at mining time.

Reply:
Do you mean by the call to this method coming from `co.rsk.core.bc.PendingStateImpl#addWireTransactions`?

Reply:
And addPendingTransaction. Apparently, the best block in null, could execute executeTransaction (in addPendingTransaction) after this validation returns true